tests/Tests.AzureAppConfiguration/FeatureManagementTests.cs

```diff
@@ -85,6 +84,111 @@ public class FeatureManagementTests
             contentType: FeatureManagementConstants.ContentType + ";charset=utf-8",
             eTag: new ETag("c3c231fd-39a0-4cb6-3237-4614474b92c1"));
 
+        List<ConfigurationSetting> _nullOrMissingConditionsFeatureFlagCollection = new List<ConfigurationSetting>
```

Comment:
Could you add more tests for invalid feature flags/Key vault ref/any malformed JSON values?

Reply:
Yea I just added them as I encountered things with the adapters, but I can try to be more thorough with all the scenarios I can think of

Reply:
Earlier, we could rely on `JsonSerializer` to handle invalid JSON data appropriately. But now that we're going to own all that token parsing logic, we should add our tests for validating JSON structure.

Reply:
Added a few tests to cover some of the json scenarios mentioned in the PR comments